src/nam_plugin.cpp

```diff
@@ -87,40 +124,12 @@ namespace NAM {
 
 				try
 				{
+				
 					// load model from path
-					const size_t pathlen = strlen(msg->path);
-					::DSP* model;
-
-					if (pathlen == 0 || pathlen >= MAX_FILE_NAME)
-					{
-						// avoid logging an error on an empty path.
-						// but do clear the model.
-						model = nullptr;
-					}
-					else
-					{
-						lv2_log_trace(&nam->logger, "Staging model change: `%s`\n", msg->path);
-
-						model = get_dsp(msg->path).release();
-
-						// Enable model loudness normalization
-						model->SetNormalize(true);
-
-						// Pre-run model to ensure all needed buffers are allocated in advance
-						if (const int32_t numSamples = nam->maxBufferSize)
-						{
-							float* buffer = new float[numSamples];
-
-							std::unordered_map<std::string, double> params = {};
-							model->process(&buffer, &buffer, 1, numSamples, 1.0, 1.0, params);
-							model->finalize_(numSamples);
-
-							delete[] buffer;
-						}
-					}
+					::DSP* model = nam->load_model(msg->path);
 
 					LV2SwitchModelMsg response = { kWorkTypeSwitch, {}, model };
-					memcpy(response.path, msg->path, pathlen);
+					memcpy(response.path, msg->path, strlen(msg->path));
```

Comment:
this breaks the case of `pathlen == 0 || pathlen >= MAX_FILE_NAME`.
so the `pathlen` should be kept outside of the function, maybe have the model load function only deal with valid paths.